src/tests/utests/xi_utest_time_event.c

```diff
@@ -0,0 +1,254 @@
+#include "tinytest.h"
+#include "tinytest_macros.h"
+#include "xi_tt_testcase_management.h"
+#include "xi_memory_checks.h"
+
+#include "xi_handle.h"
+#include "xi_vector.h"
+#include "xi_time_event.h"
+#include "xively.h"
+
+#ifndef XI_TT_TESTCASE_ENUMERATION__SECONDPREPROCESSORRUN
+
+#define TEST_TIME_EVENT_TEST_SIZE 64
+
+typedef xi_time_t( heap_element_generator )( int index );
+
+static xi_time_t index_generator( int index )
+{
+    return index;
+}
+
+static xi_time_t random_generator_0_1000( int index )
+{
+    XI_UNUSED( index );
+
+    return random() % 1000;
```

Comment:
I'd use BSP random instead, it may happen later that unittests will run on devices.